### PR TITLE
Initial Project Definition

### PR DESCRIPTION
**Pull request by elainethale**
This PR includes:
- First draft of project dimensions
- First draft of supplemental geography dimensions
- Example of how project specifies an expected dataset and its contents

### CODE REVIEW

File: project/dimensions/enduses.csv [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+natural_gas_hot_tub_heater,Hot Tub Heater,natural_gas,MBtu
+natural_gas_interior_equipment,Interior Equipment,natural_gas,MBtu
+natural_gas_lighting,Lighting,natural_gas,MBtu
+natural_gas_pool_heater,Pool Heater,natural_gas,MBtu
+natural_gas_water_systems,Water Systems,natural_gas,MBtu
+propane_clothes_dryer,Clothes Dryer,propane,MBtu
+propane_cooking_range,Cooking Range,propane,MBtu
+propane_heating,Heating,propane,MBtu
+propane_water_systems,Water Systems,propane,MBtu
+wood_heating,Heating,wood,MBtu
```
Comment (elainethale) [inline project/dimensions/enduses.csv:57]: These are the end uses dsgrid currently has for ResStock and ComStock. Are these accurate for DECARB buildings?

File: project/dimensions/enduses.csv [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+electricity_ev_dcfc,EV DCFC,electricity,MWh
+electricity_ev_l1l2,EV L1/L2,electricity,MWh
```
Comment (elainethale) [inline project/dimensions/enduses.csv:59]: These are the end uses dsgrid currently has for TEMPO LDV profiles. What should be added/changed?

File: project/dimensions/enduses.csv [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+electricity_facility_lighting,Facility Lighting,electricity,MWh
+electricity_other_facility_support,Other Facility Support,electricity,MWh
+natural_gas_other_facility_support,Other Facility Support,natural_gas,MBtu
+electricity_onsite_transportation,Onsite Transportation,electricity,MWh,
+fuel_oil_onsite_transportation,Onsite Transportation,fuel_oil,MBtu
+natural_gas_onsite_transportation,Onsite Transportation,natural_gas,MBtu
+hgl_onsite_transportation,Onsite Transportation,hgl,MBtu
+electricity_other_nonprocess,Other Nonprocess,electricity,MWh
+fuel_oil_other_nonprocess,Other Nonprocess,fuel_oil,MBtu
+natural_gas_other_nonprocess,Other Nonprocess,natural_gas,MBtu
```
Comment (elainethale) [inline project/dimensions/enduses.csv:94]: I pulled these from MECS 2018: https://www.eia.gov/consumption/manufacturing/data/2018/pdf/Table5_5.pdf. Are they accurate? Which fuels does industry plan to include?

File: project/dimensions/model_years.csv [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+2020,2020
+2022,2022
+2024,2024
+2026,2026
+2028,2028
+2030,2030
+2035,2035
+2040,2040
+2045,2045
+2050,2050
```
Comment (elainethale) [inline project/dimensions/model_years.csv:11]: These years generally align with Cambium. (Although Cambium only starts at 2024.)

File: project/dimensions/subsectors.csv [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+phev_midsize,PHEV Midsize
+phev_pickup,PHEV Pickup
+phev_suv,PHEV SUV
+bev_light_medium,BEV Light-Medium
+bev_medium,BEV Medium
+bev_non_freight_truck,BEV Non-freight truck
+bev_heavy,BEV Heavy
+bev_school_bus,BEV School Bus
+bev_transit_bus,BEV Transit Bus
+light_rail,Light Rail
```
Comment (elainethale) [inline project/dimensions/subsectors.csv:17]: Transportation, please revise. I guesstimated these from the current TEMPO dataset and our DECARB Task 6 approach document.

File: project/dimensions/subsectors.csv [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+naics_326,Plastics and Rubber Products Manufacturing
+naics_327,Nonmetallic Mineral Product Manufacturing
+naics_331,Primary Metal Manufacturing
+naics_332,Fabricated Metal Product Manufacturing
+naics_333,Machinery Manufacturing
+naics_334,Computer and Electronic Product Manufacturing
+naics_335,"Electrical Equipment, Appliance, and Component Manufacturing"
+naics_336,Transportation Equipment Manufacturing
+naics_337,Furniture and Related Product Manufacturing
+naics_339,Miscellaneous Manufacturing
```
Comment (elainethale) [inline project/dimensions/subsectors.csv:38]: Industry, please revise. I started here, with the 3-digit manufacturing NAICS codes, based on the DECARB Task 6 approach document.

File: project/project.json5 [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+                    },
+                    metric: {
+                        supplemental: {
+                            name: 'building-end-uses-collapsed',
+                            record_ids: ['building_end_uses'],
+                        },
+                    }
+                }
+            }
+        },
```
Comment (elainethale) [inline project/project.json5:26]: This specification conveys that the buildings dataset is expected to provide information either directly or through mappings for all project dimension elements except for the ones excluded by these filters on sector, subsector, and metric. That is, we expect to have (mapped) buildings data for all counties, model years, scenarios, and weather years; but only expect buildings data for a subset of the sectors ('res' and 'com'), subsectors ('unspecified'), and end-uses (project end-uses that map into 'building_end_uses').

File: project/project.json5 [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+            },
+            {
+                type: 'weather_year',
+                class: 'WeatherYear',
+                name: '2018 Weather Year',
+                display_name: 'Weather 2018',
+                description: 'Weather year, 2018 only',
+                file: 'dimensions/weather_years.csv',
+                module: 'dsgrid.dimension.standard',
+            },
```
Comment (elainethale) [inline project/project.json5:112]: Expecting timestamps and weather to follow calendar year 2018, as seen from the EST timezone. dsgrid can perform end/beginning-of-year time-wrapping and shifting from period-ending to period-beginning timestamps as needed to ingest datasets. We also support TEMPO's representative time. Please let us know if you have any time conversion questions/other time formats we need to support.

File: project/dimensions/subsectors.csv (hunk 5 above)
Comment (bborlaug) [reply project/dimensions/subsectors.csv:17]: These look good!

File: project/dimensions/enduses.csv (hunk 2 above)
Comment (bborlaug) [reply project/dimensions/enduses.csv:59]: need to breakout by electricity type (residential, commercial)

**Review by bborlaug — DISMISSED** [review]
End uses for transport should be:

```
+ electricity_ev_commercial_dcfc,EV commercial DCFC,electricity,MWh
+ electricity_ev_residential_l1l2,EV residential L1/L2,electricity,MWh
+ electricity_ev_commercial_l1l2,EV commercial L1/L2,electricity,MWh
+ electricity_lightrail,Light Rail,electricity,MWh
```

File: project/dimensions/enduses.csv (hunk 1 above)
Comment (asatremeloy) [reply project/dimensions/enduses.csv:57]: Scout's end uses match those used in the AEO and are much more aggregated than those in ResStock and ComStock. You can see Scout's residential and commercial end uses [here](https://scout-bto.readthedocs.io/en/latest/ecm_reference.html#end-use). We have developed a mapping file that maps the Stock end uses to those used in Scout, but I don't think we plan to disaggregate Scout's end-use resolution back to the Stock end uses (after developing the hourly projections), so the expected data format would instead be Scout's more limited set of end uses (and only the electric ones). Can discuss if more clarification is needed.

File: project/dimensions/model_years.csv (hunk 4 above)
Comment (asatremeloy) [reply project/dimensions/model_years.csv:11]: As we've discussed, for buildings we plan to produce results for 2035 and 2050. At one point we discussed needing to produce results for a base year (e.g., 2020 or 2025) — is that not your expectation @elainethale ?

File: project/project.json5 (hunk 7 above)
Comment (asatremeloy) [reply project/project.json5:26]: Thanks — this looks correct to me.

File: project/project.json5 (hunk 8 above)
Comment (asatremeloy) [reply project/project.json5:112]: For buildings we will be using the time conventions from ResStock/ComStock EUSS, so hopefully that doesn't require additional formats/conversions.

**Review by asatremeloy — CHANGES_REQUESTED** [review]
I added comments throughout — the one to address before merging is the building end uses comment. I'm not sure if you want to change those here or instead keep the ResStock/ComStock end uses and then pull in a separate mapping file (which we have) that maps those end uses to the Scout ones. But the dataset we will register will have the end-use resolution I linked to in my comment.

File: project/dimensions/model_years.csv (hunk 4 above)
Comment (elainethale) [reply project/dimensions/model_years.csv:11]: @asatremeloy, that is my expectation. We'll just need to specify factors that interpolate your results to these years so we can provide data to off-takers.

**Comment by elainethale** [comment]
> I added comments throughout — the one to address before merging is the building end uses comment. I'm not sure if you want to change those here or instead keep the ResStock/ComStock end uses and then pull in a separate mapping file (which we have) that maps those end uses to the Scout ones. But the dataset we will register will have the end-use resolution I linked to in my comment.

Thanks, Aven! I've aligned our end-uses with Scout's. We could add your mapping to the Stock model end-uses as a supplemental dimension if that would be helpful for any data off-takers, but that's an easy add later (no need to decide/take action now).

**Comment by asatremeloy** [comment]
Thanks @elainethale . I hope it's not too difficult to update these end uses because we will be iterating on our preliminary Task 6 buildings results with a few reviewers to determine how resolved they should be. So the list will definitely change before our data is shared with NREL, but I can keep you updated as to what the end uses are (and I can also provide a supplementary mapping file, which we already have). Thanks.

**Comment by elainethale** [comment]
> Thanks @elainethale . I hope it's not too difficult to update these end uses because we will be iterating on our preliminary Task 6 buildings results with a few reviewers to determine how resolved they should be. So the list will definitely change before our data is shared with NREL, but I can keep you updated as to what the end uses are (and I can also provide a supplementary mapping file, which we already have). Thanks.

Definitely not hard to update, @asatremeloy, especially if we can finalize by the end of October or so.

File: project/project.json5 [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+                    }
+                }
+            }
+        },
+    ],
+    dimensions: {
+        base_dimensions: [
+            {
+                type: 'geography',
+                class: 'county',
```
Comment (daniel-thom) [inline project/project.json5:34]: `County`

File: project/project.json5 [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+                name: 'End Uses by Sector',
+                display_name: 'end_uses_by_sector',
+                description: 'Selects end uses by sector, with buildings split into residential and commercial.',
+                type: 'metric',
+                file: 'dimensions/subset/end_uses_by_sector.csv',
+                create_supplemental_dimensions: false,
+                selectors: [
+                    {
+                        name: 'residential_end_uses',
+                        descripton: 'Residential End Uses',
```
Comment (daniel-thom) [inline project/project.json5:127]: `description`, and all other cases

File: project/project.json5 [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+            },
+        ],
+        subset_dimensions: [
+            {
+                name: 'End Uses by Sector',
+                display_name: 'end_uses_by_sector',
+                description: 'Selects end uses by sector, with buildings split into residential and commercial.',
+                type: 'metric',
+                file: 'dimensions/subset/end_uses_by_sector.csv',
+                create_supplemental_dimensions: false,
```
Comment (daniel-thom) [inline project/project.json5:123]: `create_supplemental_dimension` (this changed yesterday)

File: project/project.json5 [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+                        descripton: 'Residential End Uses',
+                        column_values: {'fuel_id': 'electricity', 'unit': 'MWh'},
+                    },
+                    {
+                        name: 'commercial_end_uses',
+                        descripton: 'Commercial End Uses',
+                        column_values: {'fuel_id': 'electricity', 'unit': 'MWh'},
+                    },
+                    {
+                        name: 'transportation_ldv_end_uses',
```
Comment (daniel-thom) [inline project/project.json5:136]: The records file has `transportation_end_uses` instead of these two.

File: project/project.json5 [hunk 13]
```diff
@@ -0,0 +1,255 @@
...
+            {
+                description: 'Maps US Counties 2020 L48 to CensusRegion',
+                file: 'dimension_mappings/lookup_county_to_censusregion.csv',
+                mapping_type: 'many_to_one_aggregation',
+                from_dimension: {
+                    name: 'US Counties 2020 L48',
+                    type: 'geography',
+                },
+                to_dimension: {
+                    name: 'US Census Regions',
```
Comment (daniel-thom) [inline project/project.json5:236]: `Census Region`

**Review by asatremeloy — APPROVED** [review]

**Comment by bborlaug** [comment]
@elainethale - can you add @ahcyip for review? He'll be leading the transport upload.

**Pull request merged**